Add brief discussion of contexts in the draft (#63)
panrg/path-properties: draft-irtf-panrg-path-properties.md

The terminology defined in this document is intended to be general and applicable to existing and future path-aware technologies.
Using this terminology, a path-aware technology can define and consider specific path elements and path properties on a specific level of abstraction.
For instance, a technology may define path elements as IP routers, e.g., in source routing ({{?RFC1940}}). Alternatively, it may consider path elements on a different layer of the Internet Architecture ({{?RFC1122}}) or as a collection of entities not tied to a specific layer, such as an AS or an ERO.

> Even within a single path-aware technology, terminology might differ depending on the context in which it is used.

Reviewer: Is it really the terminology that differs, or is it the specific definition you get when using this terminology, i.e., what specific component gets defined as a node/path/endpoint? (At least that's what the prior few sentences suggest, as I read them.)

I'm wondering if this should say:
```suggestion
Even within a single path-aware technology, specific definitions might differ depending on the context in which they are used.
```

> For example, the start or end node might be the communicating host in the data-plane context, and the AS of the respective host in the control-plane context.

Reviewer: If our endpoint definition lands, we can change this to "endpoint".

However, I'm not quite sure what "the data-plane context" and "the control-plane context" actually is. We do define data plane and control plane as part of the entity definition, but I still don't think these are necessarily clearly defined contexts. At least in my mind, a "control-plane context" could just as easily refer to routing between individual hosts and not just be on an AS level.

An easy fix would be to make it sound less like there is *the* single clearly defined "control plane context":

```suggestion
For example, the endpoints might be the communicating hosts in a data-plane context, and the AS of the respective hosts in a control-plane context.
```

Or maybe we can change the example:

```suggestion
For example, the endpoints might be the communicating hosts in the context of the transport layer, ASes that contain the hosts in the context of routing, or specific applications in the context of the application layer.
```

However, at this point, it feels like we're almost replicating the part of the Path definition which talks about levels of abstraction:

> Path: […] Different entities may have different visibility of a path and/or treat path elements at different levels of abstraction. For example, a path may be given as a sequence of physical nodes and the links connecting these nodes, or it may be given as a sequence of logical nodes such as a sequence of ASes or an Explicit Route Object (ERO). Similarly, the representation of a path and its properties, as it is known to a specific entity, may be more complex and include details about the physical layer technology, or it may be more abstract and only consist of a specific source and destination which is known to be reachable from that source.

But maybe that's fine, as the point is important enough that it bears repeating, as we keep getting comments about it...? :slightly_smiling_face: 
(And no, I don't want this document to define "transport layer" or "routing"...)

Author: An advantage of the second proposal is that it shows that an endpoint can also be a virtual entity, such as an application, on a host (not only the host itself). Also, it does not suffer from the fact that a reader may think, there are only two contexts, i.e., data-plane vs control-plane.

I would suggest to use the second proposal (even if the text is somewhat redundant, in my opinion it still helps with understanding).

Reviewer: Ok, sounds good! Let's go with the second proposal then.
